crates/ra_parser/src/grammar/patterns.rs

```diff
@@ -36,9 +41,14 @@ pub(super) fn pattern_r(p: &mut Parser, recovery_set: TokenSet) {
         // }
         if p.at(T![...]) || p.at(T![..=]) || p.at(T![..]) {
             let m = lhs.precede(p);
+            let dots = p.current();
             p.bump();
-            atom_pat(p, recovery_set);
-            m.complete(p, RANGE_PAT);
+            if subslice_patterns && dots == T![..] && !p.at_ts(PATTERN_FIRST) {
+                m.complete(p, SUBSLICE_PAT);
```

Comment:
Hm, I think this is still not exactly what we want, as it'll match `[1..]`, which is wrong: we need to enfroce that lhs is identifier. 

I think the best solution would be to inline `pat_list` into `slice_pat` and handle `a..` specially, simliar to `..`:

```
T![..] => p.bump(),
```

Reply:
That is, special casing should happen around here: https://github.com/rust-analyzer/rust-analyzer/blob/5b19825e376b67aabbe8d5b163bf69b1acd92f04/crates/ra_parser/src/grammar/patterns.rs#L241

Reply:
Ah, yeah, I see `pat_list` is already inlined! So, we only need to pull haling of `..` out of `pattern_with_subslice_patterns` into the `while !p.at(EOF) && !p.at(T![']']) {` loop